server/src/main/java/org/opensearch/gateway/remote/ClusterStateChecksum.java

```diff
@@ -0,0 +1,419 @@
+public class ClusterStateChecksum implements ToXContentFragment, Writeable {
+
+    private static final String ROUTING_TABLE_CS = "routing_table";
+    private static final String NODES_CS = "discovery_nodes";
+    private static final String BLOCKS_CS = "blocks";
+    private static final String CUSTOMS_CS = "customs";
+    private static final String COORDINATION_MD_CS = "coordination_md";
+    private static final String SETTINGS_MD_CS = "settings_md";
+    private static final String TRANSIENT_SETTINGS_MD_CS = "transient_settings_md";
+    private static final String TEMPLATES_MD_CS = "templated_md";
+    private static final String CUSTOM_MD_CS = "customs_md";
+    private static final String HASHES_MD_CS = "hashes_md";
+    private static final String INDICES_CS = "indices_md";
+    private static final String CLUSTER_STATE_CS = "cluster_state";
+    private static final Logger logger = LogManager.getLogger(ClusterStateChecksum.class);
+
+    long routingTableChecksum;
+    long nodesChecksum;
+    long blocksChecksum;
+    long clusterStateCustomsChecksum;
+    long coordinationMetadataChecksum;
+    long settingMetadataChecksum;
+    long transientSettingsMetadataChecksum;
+    long templatesMetadataChecksum;
+    long customMetadataMapChecksum;
+    long hashesOfConsistentSettingsChecksum;
+    long indicesChecksum;
+    long clusterStateChecksum;
+
+    public ClusterStateChecksum(ClusterState clusterState) {
+        try (
+            BytesStreamOutput out = new BytesStreamOutput();
+            BufferedChecksumStreamOutput checksumOut = new BufferedChecksumStreamOutput(out)
+        ) {
+            clusterState.routingTable().writeTo(checksumOut);
+            routingTableChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.nodes().writeTo(checksumOut);
+            nodesChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.coordinationMetadata().writeTo(checksumOut);
+            coordinationMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            Settings.writeSettingsToStream(clusterState.metadata().persistentSettings(), checksumOut);
+            settingMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            Settings.writeSettingsToStream(clusterState.metadata().transientSettings(), checksumOut);
+            transientSettingsMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.metadata().templatesMetadata().writeTo(checksumOut);
+            templatesMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.metadata().customs().forEach((key, custom) -> {
+                try {
+                    custom.writeTo(checksumOut);
+                } catch (IOException e) {
+                    logger.error("Failed to create checksum for custom metadata.", e);
```

Comment:
nit: use ParameterisedMessage here with lambda invocation.

Reply:
AFAIK, use of ParameterizedMessage is suggested when using objects in logger message, since that handles conversion better, otherwise it is cheaper to use it this way when there are no parameters to be parsed. Do you think using ParameterizedMessage would give any benefit here?